test/utils/ktesting/must.go

```diff
@@ -0,0 +1,35 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ktesting
+
+// Must takes a value and an error (or, equivalently, a single function call returning a
+// value and an error), asserts that the error did not occur, and returns the value. Use
+// this to discard "can't happen" errors that would otherwise complicate test setup, and
+// that would indicate a bug in the test case itself rather than a bug in the code being
+// tested if they occurred:
+//
+//    {
+//            description: "empty input returns first IP in range",
+//            input:       "",
+//            expected:    ktesting.Must(netip.ParseAddr("10.0.0.1")),
+//    }
+func Must[T any](val T, err error) T {
```

Comment:
I like this. As seen in the test changes, this has the potential to simplify tests.

That Go "passes through" multiple return values of a function when calling another is indeed an odd language design choice, but very useful here.

One question worth considering: can we add this helper somewhere else, ideally in staging?

It's not clear if or when `ktesting` can  become usable in tests of code in staging. If we had a better place for `Must`, we could use it there already now.

I don't have a good proposal, though.

Reply:
`verify` failed because apparently we're not allowed to include `ktesting` from everywhere in k/k?

And yes, there are places in staging that could use it. And also, it's not _just_ for testing; there are probably places outside of tests that could use it for initializing global variables from constant values too. (I know `regexp.MustCompile` is definitely used in that way.) So maybe not in ktesting. `k8s.io/utils`? It didn't seem to belong in any of the existing utils sub-packages though...

Reply:
> verify failed because apparently we're not allowed to include ktesting from everywhere in k/k?

That's possible.

> So maybe not in ktesting. k8s.io/utils? It didn't seem to belong in any of the existing utils sub-packages though...

It could be directly in `k8s.io/utils`? `utils.Must` doesn't look too bad to me.